src/SmoFacade/Database.cs

```diff
@@ -50,6 +50,29 @@ public void DropAssociatedLogins()
         });
     }
 
+    public User AddUser(UserProperties userProperties)
+    {
+      var user = Users.SingleOrDefault(u => u.Name.Equals(userProperties.Name, StringComparison.InvariantCultureIgnoreCase));
+      if (user == null) {
+        var smoUser = new Microsoft.SqlServer.Management.Smo.User(_database, userProperties.Name)
+          { Login = userProperties.LoginName };
+        smoUser.Create();
+        user = new User(smoUser, _server, this);
+
+        foreach (var role in userProperties.Roles) user.AddRole(role);
+        user.GrantPermission(userProperties.Permissions);
+      }
```

Comment:
Can you code format this?  Inconsistent use of `{}` on new lines, same lines, or none